packages/firebase_auth/firebase_auth/android/src/main/java/io/flutter/plugins/firebase/auth/FlutterFirebaseAuthPlugin.java

```diff
@@ -1573,4 +1688,145 @@ private void removeEventListeners() {
     }
     streamHandlers.clear();
   }
+
+  // Map an app id to a map of user id to a MultiFactorUser object.
+  private final Map<String, Map<String, MultiFactor>> multiFactorUserMap = new HashMap<>();
+
+  // Map an id to a MultiFactorSession object.
+  private final Map<String, MultiFactorSession> multiFactorSessionMap = new HashMap<>();
+
+  // Map an id to a MultiFactorSession object.
+  private final Map<String, MultiFactorResolver> multiFactorResolverMap = new HashMap<>();
+
+  private MultiFactor getAppMultiFactor(
+      @NonNull String appName, GeneratedAndroidFirebaseAuth.Result result) {
+    final FirebaseUser currentUser = getCurrentUser(appName);
+    if (currentUser == null) {
+      result.error(new FirebaseNoSignedInUserException("No user is signed in"));
```

Comment:
if there's an error, do we want to continue with the function?

Reply:
Good point, I'll fix that